src/Admin/SharedBlockAdmin.php

```diff
@@ -38,14 +38,19 @@ protected function generateBaseRouteName(bool $isChildAdmin = false): string
         return sprintf('%s_%s', parent::generateBaseRouteName($isChildAdmin), 'shared');
     }
 
-    protected function configureQuery(ProxyQueryInterface $query): ProxyQueryInterface
+    /**
+     * @param ORMProxyInterface<PageBlockInterface> $query
```

Comment:
You cannot add it to phpdoc for contravariance issues.

Reply:
I was ondering, why do we need this check

`\assert($query instanceof ORMProxyInterface);`, if the ProxyInterface has everything to make this method works

Reply:
The ProxyQueryInteface doesn't have getQueryBuilder ; it only works for the ORM one.

Like I said, it's an issue with psalm.
Just add PsalmSuppress MoreSpecificReturnType, LessSpecificReturnStatement with a like to https://github.com/vimeo/psalm/issues/8429

Reply:
Ok But What do I do until this issue be fixed? should I ignore this for now, and remove the comment after it be fixed there?

Reply:
The psalm suppress will ignore the error

Reply:
what do you mean  "psalm suppress"